python_modules/dagster/dagster/_core/definitions/data_version.py

```diff
@@ -44,158 +45,177 @@ def foo(x):
             value=check.str_param(value, "value"),
         )
 
 
-DEFAULT_LOGICAL_VERSION: Final[LogicalVersion] = LogicalVersion("INITIAL")
-NULL_LOGICAL_VERSION: Final[LogicalVersion] = LogicalVersion("NULL")
-UNKNOWN_LOGICAL_VERSION: Final[LogicalVersion] = LogicalVersion("UNKNOWN")
+DEFAULT_DATA_VERSION: Final[DataVersion] = DataVersion("INITIAL")
+NULL_DATA_VERSION: Final[DataVersion] = DataVersion("NULL")
+UNKNOWN_DATA_VERSION: Final[DataVersion] = DataVersion("UNKNOWN")
 
 
-class LogicalVersionProvenance(
+class DataProvenance(
     NamedTuple(
-        "_LogicalVersionProvenance",
+        "_DataProvenance",
         [
             ("code_version", str),
-            ("input_logical_versions", Mapping["AssetKey", LogicalVersion]),
+            ("input_data_versions", Mapping["AssetKey", DataVersion]),
         ],
     )
 ):
     """(Experimental) Provenance information for an asset materialization.
 
     Args:
         code_version (str): The code version of the op that generated a materialization.
-        input_logical_versions (Mapping[AssetKey, LogicalVersion]): The logical versions of the
-            inputs used for the materialization.
+        input_data_versions (Mapping[AssetKey, DataVersion]): The data versions of the
+            inputs used to generate a materialization.
     """
 
     def __new__(
         cls,
         code_version: str,
-        input_logical_versions: Mapping["AssetKey", LogicalVersion],
+        input_data_versions: Mapping["AssetKey", DataVersion],
     ):
         from dagster._core.definitions.events import AssetKey
 
-        return super(LogicalVersionProvenance, cls).__new__(
+        return super(DataProvenance, cls).__new__(
             cls,
             code_version=check.str_param(code_version, "code_version"),
-            input_logical_versions=check.mapping_param(
-                input_logical_versions,
-                "input_logical_versions",
+            input_data_versions=check.mapping_param(
+                input_data_versions,
+                "input_data_versions",
                 key_type=AssetKey,
-                value_type=LogicalVersion,
+                value_type=DataVersion,
             ),
         )
 
     @staticmethod
-    def from_tags(tags: Mapping[str, str]) -> Optional[LogicalVersionProvenance]:
+    def from_tags(tags: Mapping[str, str]) -> Optional[DataProvenance]:
         from dagster._core.definitions.events import AssetKey
 
-        code_version = tags.get(CODE_VERSION_TAG_KEY)
+        code_version = tags.get(CODE_VERSION_TAG)
         if code_version is None:
             return None
-        start_index = len(INPUT_LOGICAL_VERSION_TAG_KEY_PREFIX) + 1
-        input_logical_versions = {
-            AssetKey.from_user_string(k[start_index:]): LogicalVersion(tags[k])
+        input_data_versions = {
+            # Everything after the 2nd slash is the asset key
+            AssetKey.from_user_string(k.split("/", maxsplit=2)[-1]): DataVersion(tags[k])
```

Comment:
why the change in logic here?

Reply:
Because there are now two possible tag prefixes so we can't use the length.

Reply:
ah i see